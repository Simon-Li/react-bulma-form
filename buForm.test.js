import React from 'react';
import BuForm from './build/buForm';
import renderer from 'react-test-renderer';

// form metadata
const metadata = [
    [{
        name: 'a1',
        type: 'text',
        label: 'a1 label',
        css: 'a1-css',
        placeholder: 'a1 placeholder'
    }, {
        name: 'a2',
        type: 'text',
        label: 'a2 label',
        css: 'a2-css',
        placeholder: 'a2 placeholder'
    }],
    [{
        name: 'b1',
        type: 'checkbox',
        label: 'b1 label',
        text: 'checkbox for ...',
        css: 'b1-css'
    }, {
        name: 'b2',
        type: 'select',
        label: 'b2 label',
        options: [{ label: 'aaa', value: 'aaa' }, { label: 'bbb', value: 'bbb' }, { label: 'ccc', value: 'ccc' }]
    }],
/*
    [{
        name: 'c1',
        type: 'datepicker',
        label: 'c1 label',
        css: 'c1-css'
    }],
*/
    [{
        name: 'd1',
        type: 'textarea',
        label: 'd1 label',
        css: 'd1-css',
        placeholder: 'd1 placeholder'
    }],
    [{
        name: 'e1',
        type: 'radio',
        label: 'e1 label',
        css: 'e1-css',
        options: [{ label: 'e1-1', name: 'e1-1' }, { label: 'e1-2', name: 'e1-2' }, { label: 'e1-3', name: 'e1-3' }]
    }, {
        name: 'e2',
        type: 'radio',
        label: 'e2 label',
        css: 'e2-css',
        options: [{ label: 'e2-1', name: 'e2-1' }, { label: 'e2-2', name: 'e2-2' }, { label: 'e2-3', name: 'e2-3' }]
    }],
];
// form data from server
const data = {
    a1: 'hell world1',
    a2: 'hell world2',
    b1: true,
    b2: 'bbb',
    c1: '',
    d1: 'hell world3',
    'e1-1': true,
    'e2-2': true
};

it('renders correctly', () => {
    const tree = renderer
      .create(<BuForm name={'testForm'} metadata={metadata} data={data} alignment={'horizontal'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});
