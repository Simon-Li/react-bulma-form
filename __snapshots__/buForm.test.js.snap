// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  name="testForm"
>
  <div
    className="columns"
  >
    <div
      className="column"
    >
      <div
        className="field is-horizontal"
      >
        <div
          className="field-label"
        >
          <label
            className="label"
          >
            label
          </label>
        </div>
        <div
          className="field-body"
        >
          <div
            className="field"
          >
            <div
              className="control"
            >
              <input
                className="input undefined"
                name="input1"
                onChange={[Function]}
                placeholder="placeholder..."
                type="text"
                value="input1 text"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="column"
    >
      <div
        className="field is-horizontal"
      >
        <div
          className="field-label"
        >
          <label
            className="label"
          >
            label
          </label>
        </div>
        <div
          className="field-body"
        >
          <div
            className="field"
          >
            <div
              className="control"
            >
              <input
                className="input undefined"
                name="input2"
                onChange={[Function]}
                placeholder="placeholder"
                type="text"
                value="input2 text"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="columns"
  >
    <div
      className="column"
    >
      <div
        className="field is-horizontal"
      >
        <div
          className="field-label"
        >
          <label
            className="label"
          >
            label
          </label>
        </div>
        <div
          className="field-body"
        >
          <div
            className="field"
          >
            <div
              className="control"
            >
              <label
                className="checkbox"
              >
                <input
                  className={undefined}
                  name="checkbox1"
                  onChange={[Function]}
                  type="checkbox"
                  value={true}
                />
                check for..
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="column"
    >
      <div
        className="field is-horizontal"
      >
        <div
          className="field-label"
        >
          <label
            className="label"
          >
            label
          </label>
        </div>
        <div
          className="field-body"
        >
          <div
            className="field"
          >
            <div
              className="control"
            >
              <div
                className="select is-fullwidth"
              >
                <select
                  className={undefined}
                  name="select1"
                  onChange={[Function]}
                  value="sel2"
                >
                  <option
                    value="sel1"
                  >
                    sel1
                  </option>
                  <option
                    value="sel2"
                  >
                    sel2
                  </option>
                </select>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="columns"
  >
    <div
      className="column"
    >
      <div
        className="field is-horizontal"
      >
        <div
          className="field-label"
        >
          <label
            className="label"
          >
            label
          </label>
        </div>
        <div
          className="field-body"
        >
          <div
            className="field"
          >
            <div
              className="control"
            >
              <textarea
                className="textarea undefined"
                name="textarea1"
                onChange={[Function]}
                placeholder="placeholder.."
                type="text"
                value="textarea1 text"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="columns"
  >
    <div
      className="column"
    >
      <div
        className="field is-horizontal"
      >
        <div
          className="field-label"
        >
          <label
            className="label"
          >
            label
          </label>
        </div>
        <div
          className="field-body"
        >
          <div
            className="field"
          >
            <div
              className="control"
            >
              <label
                className="radio"
              >
                <input
                  className={undefined}
                  name="radio1"
                  onChange={[Function]}
                  type="radio"
                  value={true}
                />
                radio-1
              </label>
              <label
                className="radio"
              >
                <input
                  className={undefined}
                  name="radio1"
                  onChange={[Function]}
                  type="radio"
                  value={undefined}
                />
                radio-2
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="column"
    >
      <div
        className="field is-horizontal"
      >
        <div
          className="field-label"
        >
          <label
            className="label"
          >
            label
          </label>
        </div>
        <div
          className="field-body"
        >
          <div
            className="field"
          >
            <div
              className="control"
            >
              <label
                className="radio"
              >
                <input
                  className={undefined}
                  name="radio2"
                  onChange={[Function]}
                  type="radio"
                  value={true}
                />
                radio-3
              </label>
              <label
                className="radio"
              >
                <input
                  className={undefined}
                  name="radio2"
                  onChange={[Function]}
                  type="radio"
                  value={undefined}
                />
                radio-4
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
